### ISSUE TITLE
Compiler reports multiple *different* errors for missing 'Deconstruct' methods.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#15381]
Repro case:

```c#
class C
{ 
    void M(object o)
    {
        (int x, int y) = o;
    }
}
```

Errors reported:

```
error CS1061: 'object' does not contain a definition for 'Deconstruct' and no extension method 'Deconstruct' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
error CS8129: No Deconstruct instance or extension method was found for type 'object', with 2 out parameters.
```

IDE would like to only have CS8129 so that we can offer to Add a using for that missing method.

Also, the experience isn't great that multiple errors are reported here.

**Comment by CyrusNajmabadi** [comment]
Tagging @gafter

**Comment by CyrusNajmabadi** [comment]
@gafter @jaredpar This was opened 19 days ago, and does not seem to have been triaged at all.

**Comment by jcouv** [comment]
Just to confirm, I assume that this issue is not blocking, since CS8129 is there.

For some context on the design, the first error is whatever diagnostics we get from `BindMethodGroupInvocation`. That covers all the traditional invocation problems and can offer more specific feedback than the second error. 
But they aren't specific to deconstruction, so the second error offers some useful information which can guide the user with deconstructions. 
For that reason, I wouldn't recommend just removing the first error (I think that is a worse user experience).
We could try to nest diagnostics instead, but that seems heavy compared to simply stacking them.

For reference, this logic is in `MakeDeconstructInvocationExpression`.

**Comment by CyrusNajmabadi** [comment]
> Just to confirm, I assume that this issue is not blocking, since CS8129 is there.

There seems to be an issue with the CodeFix system.  If there are multiple errors at the same position and span, it can really screw things up.

**Comment by jcouv** [comment]
@CyrusNajmabadi Let me know the urgency on this, or whether the code fixer infrastructure can now tolerate two diagnostics on the same span.
I may push out of 15.3 unless pressing IDE dependency.

**Comment by CyrusNajmabadi** [comment]
There is zero urgency.  IDE worked around the issue a while back.
